src/gui/sdl_mapper.cpp

```diff
@@ -3246,21 +3246,49 @@ static void MAPPER_Destroy(Section *sec) {
 	SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 }
 
-void MAPPER_BindKeys(Section *sec)
+static bool should_skip_bind_keys(const Section_prop* section)
 {
-	// Get the mapper file set by the user
-	const auto section = static_cast<const Section_prop *>(sec);
-	const auto mapperfile_value = section->Get_string("mapperfile");
-	const auto property = section->Get_path("mapperfile");
-	
-	// Filter out unneeded calls
+	// Filter out unneeded calls - we only need to execute MAPPER_BindKeys
+	// when one of the values have changed; we do not want it to be executed
+	// if 'window_titlebar' was set (even to the same value as it had) to
+	// avoid screen flicker
```

Comment:
Dunno, that's a lot of effort for a little flicker. I'd say let it flicker, or better yet, can't we flip this around and detect if `window_titlebar` is the same of has changed? This seems rather hacky and we'll need to remember to maintain this list when we add/remove `[sdl]` settings.

Well, but then the whole `[sdl]` section handling and the whole init process is hacky as hell... So dunno... I guess you don't want to refactor the whole thing as part of this 😄 I'm just wondering if there's a cleaner solution with less maintenance overhead and room for error when `[sdl]` section settings are added or renamed.

Reply:
Ideally the config system would check if the value is already set to what the user is trying to change it to and avoid re-loading the module if that is the case.  This should happen for all modules but `[sdl]` is the most obvious offender because it reloads the window and has a visual flicker.

Probably you don't want to dive into that for this regression fix though.

Reply:
Actually, it looks like the command line utility is responsible for calling the init + destroy functions.  I believe this is the only way currently that settings can be changed at runtime.  You could try adding a check here to see if the values are the same.  I'd be a little worried about regressions though.  The config system is a fickle beast.

https://github.com/dosbox-staging/dosbox-staging/blob/f55f0ae3565c144549001fe65bcf439c315fa510/src/misc/programs.cpp#L843-L855

Reply:
For now I have simplified the condition to just check for _window_titlebar_ changes - at the expense of flicker if the value wasn't actually changed.

Right, I don't want to mess with the config system right now, or to start yet another serious rework.

Reply:
Thanks @FeralChild64